traitlets/config/application.py

```diff
@@ -312,16 +312,26 @@ def print_alias_help(self):
                 classdict[c.__name__] = c
 
         for alias, longname in self.aliases.items():
-            classname, traitname = longname.split('.',1)
-            cls = classdict[classname]
-
-            trait = cls.class_traits(config=True)[traitname]
-            help = cls.class_get_trait_help(trait).splitlines()
-            # reformat first line
-            help[0] = help[0].replace(longname, alias) + ' (%s)'%longname
-            if len(alias) == 1:
-                help[0] = help[0].replace('--%s='%alias, '-%s '%alias)
-            lines.extend(help)
+            try:
+                classname, traitname = longname.split('.',1)
+                cls = classdict[classname]
+
+                trait = cls.class_traits(config=True)[traitname]
+                fhelp = cls.class_get_trait_help(trait).splitlines()
+
+                if not isinstance(alias, tuple):
+                    alias = (alias, )
+                alias = sorted(alias, key=len)
+                alias = ', '.join(('--%s' if len(m) > 1 else '-%s') % m
+                                  for m in alias)
+
+                # reformat first line
+                fhelp[0] = fhelp[0].replace(longname, alias)
+                lines.extend(fhelp)
+                lines.append(indent("Equivalent to: [%s]" % longname))
+            except Exception as ex:
+                self.log.error('Failed collecting help-message for aias %r, due to: %s',
```

Comment:
While developing it, the stack-traces for such errors happened very early in the boot-strap process and it was extremely confusing for me to debug such issues (practically speaking); eventually I had to manually wrap the offending code in a try-catch every time; eventually, left it there for others, as explained in the commit message.

You can try it yourself if you have some class without `description` trait, or badly formed flag.
Maybe we can retrofit the error behavior to help developers.